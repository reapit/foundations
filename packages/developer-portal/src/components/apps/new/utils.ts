import { Dispatch, SetStateAction } from 'react'
import { AppAuthFlow, AppNewStepId } from './config'
import { AppWizardState } from '../state/use-app-state'

export const handleSetAppWizardState =
  (
    setAppWizardState: Dispatch<SetStateAction<AppWizardState>>,
    nextStep: AppNewStepId,
    authFlow: AppAuthFlow = 'clientCredentials',
  ) =>
  () => {
    setAppWizardState((currentState) => ({
      ...currentState,
      nextStep,
      authFlow,
    }))
  }
