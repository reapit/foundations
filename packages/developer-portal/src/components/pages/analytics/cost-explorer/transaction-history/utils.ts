import { Dayjs } from 'dayjs'

export const MAX_NUMBER_TRANSACTION_FIRST_PAGE = 3
export const MAX_NUMBER_TRANSACTION_PER_PAGE = 6

export const formatRenderDate = (month: Dayjs) => month.format('MMMM YYYY')
export const formatRequestDate = (month: Dayjs) => month.format('YYYY-MM')
export const generatePreviousTransactionDate = ({
  currentDate,
  developerCreateDate,
}: {
  currentDate: Dayjs
  developerCreateDate: Dayjs
}) => {
  const dates: Dayjs[] = []
  let tempDate = currentDate

  while (developerCreateDate.isBefore(tempDate) && developerCreateDate.month() !== currentDate.month()) {
    tempDate = tempDate.add(-1, 'M')

    dates.push(tempDate)
  }

  return dates
}
