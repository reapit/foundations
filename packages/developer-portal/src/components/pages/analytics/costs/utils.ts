import { BillingBreakdownForMonthV2Model, ServiceItemBillingV2Model } from '@reapit/foundations-ts-definitions'
import dayjs from 'dayjs'
import { AnalyticsFilterState } from '../state/use-analytics-state'

export const getMonthsRange = (analyticsFilterState: AnalyticsFilterState, format: string): string[] => {
  const { monthFrom, monthTo } = analyticsFilterState
  const numberMonths = dayjs(monthTo).diff(dayjs(monthFrom), 'month') + 1
  const monthRequests: string[] = []

  for (let i = 0; i < numberMonths; i++) {
    monthRequests.push(dayjs(monthTo).subtract(i, 'month').format(format))
  }

  return monthRequests
}

export const aggregateItems = (items: ServiceItemBillingV2Model[]): ServiceItemBillingV2Model[] =>
  items.reduce<ServiceItemBillingV2Model[]>((aggregatedItems, item) => {
    const aggregatedItem = aggregatedItems.find((aggregatedItem) => {
      return aggregatedItem.name === item.name
    })

    if (!aggregatedItem) {
      aggregatedItems.push(item)
    } else {
      aggregatedItem.amount = (aggregatedItem.amount ?? 0) + (item.amount ?? 0)
      aggregatedItem.cost = (aggregatedItem.cost ?? 0) + (item.cost ?? 0)
      aggregatedItem.itemCount = (aggregatedItem.itemCount ?? 0) + (item.itemCount ?? 0)
      aggregatedItem.items = aggregateItems([...(aggregatedItem.items ?? []), ...(item.items ?? [])])
    }
    return aggregatedItems
  }, [])

export const handleAggregateBillingData = (billing: BillingBreakdownForMonthV2Model[]) => () =>
  billing.reduce(
    (billingAggregated, nextBillingItem): BillingBreakdownForMonthV2Model => {
      if (billingAggregated.totalCost && nextBillingItem.totalCost) {
        billingAggregated.totalCost = billingAggregated.totalCost + nextBillingItem.totalCost
      } else {
        billingAggregated.totalCost = nextBillingItem.totalCost
      }

      nextBillingItem.services?.forEach((service) => {
        const aggregatedService = billingAggregated.services?.find((item) => item.name === service.name)

        if (!aggregatedService) {
          billingAggregated.services?.push(service)
        } else {
          aggregatedService.amount = (aggregatedService.amount ?? 0) + (service.amount ?? 0)
          aggregatedService.cost = (aggregatedService.cost ?? 0) + (service.cost ?? 0)
          aggregatedService.itemCount = (aggregatedService.itemCount ?? 0) + (service.itemCount ?? 0)
          aggregatedService.items = aggregateItems([...(aggregatedService.items ?? []), ...(service.items ?? [])])
        }
      })

      return billingAggregated
    },
    { services: [], totalCost: 0 },
  )

export const flattenBillingData = (services: ServiceItemBillingV2Model[]) =>
  services?.reduce<(string | number | undefined)[][]>(
    (accumulator, { items, name, amount, cost, itemCount }) => {
      accumulator.push([name, itemCount, amount, cost])
      if (items?.length) {
        accumulator.push(...flattenBillingData(items))
      }
      return accumulator
    },
    [['Service Name', 'Item Count', 'Number Items', 'Cost']],
  )
