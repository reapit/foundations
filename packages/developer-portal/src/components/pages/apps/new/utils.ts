import { Dispatch, SetStateAction } from 'react'
import { AppAuthFlow, AppNewStepId } from './config'
import { AppWizardState } from './use-app-wizard'

export const handleSetAppWizardState =
  (
    setAppWizardState: Dispatch<SetStateAction<AppWizardState>>,
    nextStep: AppNewStepId,
    authFlow: AppAuthFlow = 'clientCredentials',
  ) =>
  () => {
    setAppWizardState((currentState) => ({
      ...currentState,
      nextStep,
      authFlow,
    }))
  }
