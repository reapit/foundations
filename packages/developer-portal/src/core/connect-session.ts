import { ReapitConnectBrowserSession } from '@reapit/connect-session'

// Needs to be a singleton as the class is stateful
export const reapitConnectBrowserSession = new ReapitConnectBrowserSession({
  connectClientId: process.env.connectClientId,
  connectOAuthUrl: process.env.connectOAuthUrl,
  connectLoginRedirectPath: '/apps',
  connectUserPoolId: process.env.connectUserPoolId,
  connectApplicationTimeout: 86400000, // 24hrs in ms
})
