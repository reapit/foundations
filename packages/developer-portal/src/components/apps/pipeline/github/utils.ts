import { ReapitConnectSession } from '@reapit/connect-session'

export type GithubCodeRequest = {
  code: string
  redirect_uri: string
}

export type GithubRefreshRequest = {
  redirect_uri: string
  refresh_token: string
}

export type GithubAuthorizationBody = GithubCodeRequest | GithubRefreshRequest

const githubClientId = process.env.githubClientId

export const authenticateWithGithub = async (data: GithubAuthorizationBody, connectSession: ReapitConnectSession) => {
  // TODO need to make the below domain an env
  const response = await fetch('https://deployments.dev.paas.reapit.cloud/github/auth', {
    method: 'post',
    body: JSON.stringify(data),
    headers: {
      authorization: `Bearer ${connectSession?.idToken}`,
      'content-type': 'application/json',
    },
  })

  return response.json()
}

export const redirectToGithub = (redirect_uri: string, route?: string) => {
  // TODO verify state?
  const state = Buffer.from(JSON.stringify({ route })).toString('base64')
  window.location.href = `https://github.com/login/oauth/authorize?client_id=${githubClientId}&redirect_uri=${redirect_uri}&state=${state}`
}

const connectIsDesktop = Boolean(window['__REAPIT_MARKETPLACE_GLOBALS__'])
export const storageMethod = connectIsDesktop ? window.localStorage : window.sessionStorage
