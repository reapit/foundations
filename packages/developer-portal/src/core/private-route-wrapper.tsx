import * as React from 'react'
import Menu from '@/components/ui/menu'
import { Loader, Section, FlexContainerResponsive, AppNavContainer, FlexContainerBasic } from '@reapit/elements'
import { Redirect, useLocation } from 'react-router'
import Routes from '@/constants/routes'
import { useReapitConnect } from '@reapit/connect-session'
import { reapitConnectBrowserSession } from '@/core/connect-session'
import { getCookieString, COOKIE_DEVELOPER_FIRST_TIME_LOGIN_COMPLETE } from '@/utils/cookie'

const { Suspense } = React

export type PrivateRouteWrapperProps = {
  children?: React.ReactNode
  path: string
  showMenu?: boolean
}

/* export const handleOnAcceptClientWelcome = ({
  dispatch,
  setClientTermAcceptedCookieAndState,
}: {
  dispatch: Dispatch
  setClientTermAcceptedCookieAndState: ActionCreator<boolean>
}) => () => dispatch(setClientTermAcceptedCookieAndState(true)) */

export const PrivateRouteWrapper: React.FunctionComponent<PrivateRouteWrapperProps> = ({
  children,
  showMenu = true,
}) => {
  const { connectSession, connectInternalRedirect } = useReapitConnect(reapitConnectBrowserSession)
  const location = useLocation()
  const currentUri = `${location.pathname}${location.search}`
  const isRoot = connectInternalRedirect === '/?' || connectInternalRedirect === '/' || window.location.pathname === '/'

  if (!connectSession) {
    return null
  }

  if (isRoot) {
    return <Redirect to={Routes.APPS} />
  }

  if (connectInternalRedirect && currentUri !== connectInternalRedirect) {
    return <Redirect to={connectInternalRedirect} />
  }

  const hasReadWelcome = Boolean(getCookieString(COOKIE_DEVELOPER_FIRST_TIME_LOGIN_COMPLETE))
  if (!hasReadWelcome && location.pathname === Routes.APPS) {
    return <Redirect to={Routes.WELCOME} />
  }

  return (
    <AppNavContainer>
      {showMenu && <Menu />}
      <FlexContainerBasic flexColumn isScrollable>
        <FlexContainerResponsive
          hasPadding
          flexColumn
          // I want to allow scrolling beyond the end of the page to allow for the toast notification
          // except on the Gitbook page because the iframe handles it's own scrolling
          isPageContainer={location.pathname !== Routes.API_DOCS}
        >
          <Suspense
            fallback={
              <Section>
                <Loader />
              </Section>
            }
          >
            {children}
          </Suspense>
        </FlexContainerResponsive>
      </FlexContainerBasic>
    </AppNavContainer>
  )
}

export default PrivateRouteWrapper
