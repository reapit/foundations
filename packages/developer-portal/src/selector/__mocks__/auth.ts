import { ReapitConnectSession } from '@reapit/connect-session'

export const auth: ReapitConnectSession = {
  accessToken: '1231231',
  idToken: '1231321',
  loginIdentity: {
    adminId: '1',
    clientId: '2',
    developerId: '3',
    email: '123',
    groups: ['AgencyCloudDeveloperEdition'],
    userCode: '123',
    name: '1312',
    orgName: 'Reapit Ltd',
    orgId: 'SOME_ORG_ID',
  },
  refreshToken: '131',
}
