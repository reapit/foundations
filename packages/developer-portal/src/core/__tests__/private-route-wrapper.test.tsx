import React from 'react'
import { PrivateRouteWrapper, handleRedirectRegistraitionPage } from '../private-route-wrapper'
import { render } from '../../tests/react-testing'
import { ReapitConnectSession } from '@reapit/connect-session'
import { COGNITO_GROUP_USERS } from '../../utils/auth'
import Routes from '../../constants/routes'

process.env.pipelineWhitelist = []
process.env.liveChatWhitelist = []

jest.mock('uuid', () => ({
  validate: jest.fn(() => true),
}))

jest.mock('@reapit/connect-session', () => ({
  ReapitConnectBrowserSession: jest.fn(),
  useReapitConnect: () => ({
    connectSession: {
      loginIdentity: {
        developerId: 'MOCK_DEVELOPER_ID',
      },
    },
    connectInternalRedirect: '',
  }),
}))

describe('PrivateRouteWrapper', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot', () => {
    window.location.pathname = Routes.APPS
    const wrapper = render(
      <PrivateRouteWrapper>
        <div />
      </PrivateRouteWrapper>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})

describe('handleRedirectToAuthenticationPage', () => {
  it('should redirect to authentication page for CLIENT', () => {
    const mockConnectSession = {
      loginIdentity: {
        developerId: '',
      },
    } as ReapitConnectSession
    const navigate = jest.fn()
    const curried = handleRedirectRegistraitionPage(navigate, mockConnectSession)
    curried()
    expect(navigate).toBeCalledWith(`${Routes.SELECT_ROLE}`)
  })

  it('should redirect to authentication page for DEVELOPER', () => {
    const mockConnectSession = {
      loginIdentity: {
        developerId: '',
        agencyCloudId: 'RES',
        groups: [COGNITO_GROUP_USERS],
      },
    } as ReapitConnectSession
    const navigate = jest.fn()
    const curried = handleRedirectRegistraitionPage(navigate, mockConnectSession)
    curried()
    expect(navigate).toBeCalledWith(`${Routes.CUSTOMER_REGISTER}`)
  })
})
