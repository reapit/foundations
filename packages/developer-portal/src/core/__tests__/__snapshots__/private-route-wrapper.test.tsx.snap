// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRouteWrapper should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <mock-styled.div />
      <mock-styled.main>
        <mock-styled.nav
          classname=""
        >
          <mock-styled.a
            classname="mock-css.
  @media screen and (min-width: 768px) {
    --nav-menu-icon-primary-accent: var(--color-white);
    color: var(--nav-menu-text-hover);
    background-color: var(--nav-menu-background-accent);
  }
"
          >
            <mock-styled.span
              classname="mock-css.
  font-size: 2.5rem;
"
            >
              <svg />
            </mock-styled.span>
            <mock-styled.span
              classname="mock-css. mock-css.
  margin-left: auto;
 mock-css.
  margin-right: 0.375rem;
 mock-css.
  @media screen and (min-width: 768px) {
    height: 0;
    width: 0;
    visibility: hidden;
  }
"
              style="font-size: 2rem;"
            >
              <svg />
            </mock-styled.span>
          </mock-styled.a>
          <mock-styled.a
            classname=""
          >
            <mock-styled.span
              classname="mock-css.
  @media screen and (max-width: 767px) {
    margin-left: 0.75rem;
  }
 mock-css.
  font-size: 2.5rem;
"
            >
              <svg />
            </mock-styled.span>
            Apps
          </mock-styled.a>
          <mock-styled.div>
            <mock-styled.a
              classname=""
            >
              <span>
                My Apps
              </span>
            </mock-styled.a>
            <mock-styled.a
              classname=""
            >
              <span>
                New App
              </span>
            </mock-styled.a>
          </mock-styled.div>
          <mock-styled.a
            classname=""
          >
            <mock-styled.span
              classname="mock-css.
  @media screen and (max-width: 767px) {
    margin-left: 0.75rem;
  }
 mock-css.
  font-size: 2.5rem;
"
            >
              <svg />
            </mock-styled.span>
            Analytics
          </mock-styled.a>
          <mock-styled.div>
            <mock-styled.a
              classname=""
            >
              <span>
                API Usage
              </span>
            </mock-styled.a>
            <mock-styled.a
              classname=""
            >
              <span>
                Costs
              </span>
            </mock-styled.a>
            <mock-styled.a
              classname=""
            >
              <span>
                Installations
              </span>
            </mock-styled.a>
          </mock-styled.div>
          <mock-styled.a
            classname=""
          >
            <mock-styled.span
              classname="mock-css.
  @media screen and (max-width: 767px) {
    margin-left: 0.75rem;
  }
 mock-css.
  font-size: 2.5rem;
"
            >
              <svg />
            </mock-styled.span>
            API
          </mock-styled.a>
          <mock-styled.div>
            <mock-styled.a
              classname=""
            >
              <span>
                Foundations API
              </span>
            </mock-styled.a>
            <mock-styled.a
              classname=""
            >
              <span>
                Webhooks
              </span>
            </mock-styled.a>
            <mock-styled.a
              classname=""
            >
              <span>
                GraphQL
              </span>
            </mock-styled.a>
            <mock-styled.a
              classname=""
            >
              <span>
                Desktop
              </span>
            </mock-styled.a>
          </mock-styled.div>
          <mock-styled.a
            classname=""
          >
            <mock-styled.span
              classname="mock-css.
  @media screen and (max-width: 767px) {
    margin-left: 0.75rem;
  }
 mock-css.
  font-size: 2.5rem;
"
            >
              <svg />
            </mock-styled.span>
            UI
          </mock-styled.a>
          <mock-styled.a
            classname=""
          >
            <mock-styled.span
              classname="mock-css.
  @media screen and (max-width: 767px) {
    margin-left: 0.75rem;
  }
 mock-css.
  font-size: 2.5rem;
"
            >
              <svg />
            </mock-styled.span>
            Docs
          </mock-styled.a>
          <mock-styled.div>
            <mock-styled.a
              classname=""
            >
              <span>
                APIs
              </span>
            </mock-styled.a>
            <mock-styled.a
              classname=""
            >
              <span>
                Warehouse
              </span>
            </mock-styled.a>
          </mock-styled.div>
          <mock-styled.a
            classname=""
          >
            <mock-styled.span
              classname="mock-css.
  @media screen and (max-width: 767px) {
    margin-left: 0.75rem;
  }
 mock-css.
  font-size: 2.5rem;
"
            >
              <svg />
            </mock-styled.span>
            AppMarket
          </mock-styled.a>
          <mock-styled.a
            classname="mock-css.
  @media screen and (min-width: 768px) {
    margin-top: auto;
  }
"
          >
            <mock-styled.span
              classname="mock-css.
  @media screen and (max-width: 767px) {
    margin-left: 0.75rem;
  }
 mock-css.
  font-size: 2.5rem;
"
            >
              <svg />
            </mock-styled.span>
            Settings
          </mock-styled.a>
          <mock-styled.div>
            <mock-styled.a
              classname=""
            >
              <span>
                Profile
              </span>
            </mock-styled.a>
            <mock-styled.a
              classname=""
            >
              <span>
                Password
              </span>
            </mock-styled.a>
            <mock-styled.a
              classname=""
            >
              <span>
                Members
              </span>
            </mock-styled.a>
            <mock-styled.a
              classname=""
            >
              <span>
                Company
              </span>
            </mock-styled.a>
            <mock-styled.a
              classname=""
            >
              <span>
                Subscriptions
              </span>
            </mock-styled.a>
          </mock-styled.div>
        </mock-styled.nav>
        <div />
      </mock-styled.main>
    </div>
  </body>,
  "container": <div>
    <mock-styled.div />
    <mock-styled.main>
      <mock-styled.nav
        classname=""
      >
        <mock-styled.a
          classname="mock-css.
  @media screen and (min-width: 768px) {
    --nav-menu-icon-primary-accent: var(--color-white);
    color: var(--nav-menu-text-hover);
    background-color: var(--nav-menu-background-accent);
  }
"
        >
          <mock-styled.span
            classname="mock-css.
  font-size: 2.5rem;
"
          >
            <svg />
          </mock-styled.span>
          <mock-styled.span
            classname="mock-css. mock-css.
  margin-left: auto;
 mock-css.
  margin-right: 0.375rem;
 mock-css.
  @media screen and (min-width: 768px) {
    height: 0;
    width: 0;
    visibility: hidden;
  }
"
            style="font-size: 2rem;"
          >
            <svg />
          </mock-styled.span>
        </mock-styled.a>
        <mock-styled.a
          classname=""
        >
          <mock-styled.span
            classname="mock-css.
  @media screen and (max-width: 767px) {
    margin-left: 0.75rem;
  }
 mock-css.
  font-size: 2.5rem;
"
          >
            <svg />
          </mock-styled.span>
          Apps
        </mock-styled.a>
        <mock-styled.div>
          <mock-styled.a
            classname=""
          >
            <span>
              My Apps
            </span>
          </mock-styled.a>
          <mock-styled.a
            classname=""
          >
            <span>
              New App
            </span>
          </mock-styled.a>
        </mock-styled.div>
        <mock-styled.a
          classname=""
        >
          <mock-styled.span
            classname="mock-css.
  @media screen and (max-width: 767px) {
    margin-left: 0.75rem;
  }
 mock-css.
  font-size: 2.5rem;
"
          >
            <svg />
          </mock-styled.span>
          Analytics
        </mock-styled.a>
        <mock-styled.div>
          <mock-styled.a
            classname=""
          >
            <span>
              API Usage
            </span>
          </mock-styled.a>
          <mock-styled.a
            classname=""
          >
            <span>
              Costs
            </span>
          </mock-styled.a>
          <mock-styled.a
            classname=""
          >
            <span>
              Installations
            </span>
          </mock-styled.a>
        </mock-styled.div>
        <mock-styled.a
          classname=""
        >
          <mock-styled.span
            classname="mock-css.
  @media screen and (max-width: 767px) {
    margin-left: 0.75rem;
  }
 mock-css.
  font-size: 2.5rem;
"
          >
            <svg />
          </mock-styled.span>
          API
        </mock-styled.a>
        <mock-styled.div>
          <mock-styled.a
            classname=""
          >
            <span>
              Foundations API
            </span>
          </mock-styled.a>
          <mock-styled.a
            classname=""
          >
            <span>
              Webhooks
            </span>
          </mock-styled.a>
          <mock-styled.a
            classname=""
          >
            <span>
              GraphQL
            </span>
          </mock-styled.a>
          <mock-styled.a
            classname=""
          >
            <span>
              Desktop
            </span>
          </mock-styled.a>
        </mock-styled.div>
        <mock-styled.a
          classname=""
        >
          <mock-styled.span
            classname="mock-css.
  @media screen and (max-width: 767px) {
    margin-left: 0.75rem;
  }
 mock-css.
  font-size: 2.5rem;
"
          >
            <svg />
          </mock-styled.span>
          UI
        </mock-styled.a>
        <mock-styled.a
          classname=""
        >
          <mock-styled.span
            classname="mock-css.
  @media screen and (max-width: 767px) {
    margin-left: 0.75rem;
  }
 mock-css.
  font-size: 2.5rem;
"
          >
            <svg />
          </mock-styled.span>
          Docs
        </mock-styled.a>
        <mock-styled.div>
          <mock-styled.a
            classname=""
          >
            <span>
              APIs
            </span>
          </mock-styled.a>
          <mock-styled.a
            classname=""
          >
            <span>
              Warehouse
            </span>
          </mock-styled.a>
        </mock-styled.div>
        <mock-styled.a
          classname=""
        >
          <mock-styled.span
            classname="mock-css.
  @media screen and (max-width: 767px) {
    margin-left: 0.75rem;
  }
 mock-css.
  font-size: 2.5rem;
"
          >
            <svg />
          </mock-styled.span>
          AppMarket
        </mock-styled.a>
        <mock-styled.a
          classname="mock-css.
  @media screen and (min-width: 768px) {
    margin-top: auto;
  }
"
        >
          <mock-styled.span
            classname="mock-css.
  @media screen and (max-width: 767px) {
    margin-left: 0.75rem;
  }
 mock-css.
  font-size: 2.5rem;
"
          >
            <svg />
          </mock-styled.span>
          Settings
        </mock-styled.a>
        <mock-styled.div>
          <mock-styled.a
            classname=""
          >
            <span>
              Profile
            </span>
          </mock-styled.a>
          <mock-styled.a
            classname=""
          >
            <span>
              Password
            </span>
          </mock-styled.a>
          <mock-styled.a
            classname=""
          >
            <span>
              Members
            </span>
          </mock-styled.a>
          <mock-styled.a
            classname=""
          >
            <span>
              Company
            </span>
          </mock-styled.a>
          <mock-styled.a
            classname=""
          >
            <span>
              Subscriptions
            </span>
          </mock-styled.a>
        </mock-styled.div>
      </mock-styled.nav>
      <div />
    </mock-styled.main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
